### PR TITLE
Inactive roles for subjects

### LINKED ISSUE #218: Do not remove ROLE_PARTICIPANT but make it inactive
Instead of removing the `ROLE_PARTICIPANT` role whenever a user is reassigned, it should be set to inactive. That way the REST API has a way of knowing whether a user was ever assigned to a given project. The implementation could have two parts:
1. the JWT only contains roles that are active, so participants will no longer be able to submit data for an inactive project.
2. the API exposes inactive participants, something like:
`GET /projects/radartest/participants`
```json
{
   "participants": [...],
   "inactiveParticipants": [...]
}
```

### PR DESCRIPTION
**Pull request by nivemaham**
Closes #218

### CODE REVIEW

File: radar-auth/src/main/java/org/radarcns/auth/authorization/Permissions.java [hunk 1]
```diff
@@ -115,5 +116,10 @@ private static void initPermissions() {
         Arrays.asList(Permission.SUBJECT_READ, Permission.SUBJECT_UPDATE,
                 Permission.MEASUREMENT_CREATE, Permission.MEASUREMENT_READ).stream()
                         .forEach(p -> PERMISSION_MATRIX.get(p).add(PARTICIPANT));
+
+        /* Inactive participant */
+        // Can update and read own data and can read and write own measurements
```
Comment (dennyverbeeck) [inline radar-auth/src/main/java/org/radarcns/auth/authorization/Permissions.java]: Small mistake here in comment, can not read or write measurements

File: src/main/java/org/radarcns/management/security/ClaimsTokenEnhancer.java [hunk 2]
```diff
@@ -59,7 +60,9 @@ public OAuth2AccessToken enhance(OAuth2AccessToken accessToken,
             Optional<User> optUser = userRepository.findOneByLogin(userName);
             if (optUser.isPresent()) {
                 List<String> roles = optUser.get().getRoles().stream()
-                        .filter(role -> Objects.nonNull(role.getProject()))
+                        .filter(role -> Objects.nonNull(role.getProject())
+                                && !AuthoritiesConstants.INACTIVE_PARTICIPANT.equals(role
```
Comment (dennyverbeeck) [inline src/main/java/org/radarcns/management/security/ClaimsTokenEnhancer.java]: Shouldn't we include the INACTIVE_PARTICIPANT role in the JWT anyway? If the token does not contain the role, then the user will not be able to get access to old project info based on this role.

File: src/main/java/org/radarcns/management/service/SubjectService.java [hunk 3]
```diff
@@ -152,18 +156,34 @@ public SubjectDTO updateSubject(SubjectDTO subjectDto) {
             source.setAssigned(true);
         }
         // update participant role
-        Set<Role> managedRoles = subject.getUser().getRoles().stream()
-                .filter(r -> !AuthoritiesConstants.PARTICIPANT.equals(r.getAuthority().getName()))
-                .collect(Collectors.toSet());
-        managedRoles.add(getProjectParticipantRole(subjectDto.getProject()));
+        Set<Role> managedRoles = updateParticipantRoles(subject, subjectDto);
```
Comment (dennyverbeeck) [inline src/main/java/org/radarcns/management/service/SubjectService.java]: Minor thing: I was a bit confused until i realized `subject` refers to the 'old' subject and `subjectDto` refers to the updated info. Perhaps the variable names could be `oldSubject` and `newSubject` or something along those lines.

File: src/main/java/org/radarcns/management/web/rest/SubjectResource.java [hunk 4]
```diff
@@ -220,26 +222,54 @@
...
             throws NotAuthorizedException {
         checkPermission(getJWT(servletRequest), SUBJECT_READ);
         log.debug("ProjectName {} and external {}", projectName, externalId);
+        // if not specified do not include inactive patients
+        boolean withInactive = withInactiveParticipants != null ? withInactiveParticipants : false;
         if (projectName != null && externalId != null) {
-            Optional<Subject> subject = subjectRepository
-                    .findOneByProjectNameAndExternalId(projectName, externalId);
+            Optional<Subject> subject = Optional.empty();
+            if (!withInactive) {
```
Comment (dennyverbeeck) [inline src/main/java/org/radarcns/management/web/rest/SubjectResource.java]: Also a minor thing: perhaps you can do a single check at the start of the method to create the correct list of authorities (either with or without `INACTIVE_PARTICIPANT`), and then use that list in the rest of the method, then you can remove the repeated checks on `withInactive`.

File: src/main/resources/config/liquibase/changelog/20180306165800_add_active_column_to_role.xml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd
+                        http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd">
+
+    <changeSet id="20180306165800-0" author="nivethika@thehyve.nl" >
+      <addColumn tableName="radar_role">
+        <column name="active_role" type="boolean" defaultValueBoolean="true"/>
+      </addColumn>
+    </changeSet>
+</databaseChangeLog>
```
Comment (dennyverbeeck) [inline src/main/resources/config/liquibase/changelog/20180306165800_add_active_column_to_role.xml]: Do we still need this file? It's not referenced in master.xml.

File: src/main/java/org/radarcns/management/security/ClaimsTokenEnhancer.java (hunk 2 above)
Comment (nivemaham) [reply src/main/java/org/radarcns/management/security/ClaimsTokenEnhancer.java]: I don't think so. I would consider a token to have information that could only allow them to get certain access, not the data from history that is not valid anymore.

File: src/main/java/org/radarcns/management/web/rest/SubjectResource.java (hunk 4 above)
Comment (nivemaham) [reply src/main/java/org/radarcns/management/web/rest/SubjectResource.java]: indeed. That was the idea. :( I missed it

File: src/main/resources/config/liquibase/changelog/20180306165800_add_active_column_to_role.xml (hunk 5 above)
Comment (nivemaham) [reply src/main/resources/config/liquibase/changelog/20180306165800_add_active_column_to_role.xml]: Nope. we don't.

**Pull request merged**